pkg/cloud/ibmcloud/actuators/machine/actuator.go

```diff
@@ -56,15 +59,17 @@ const (
 
 // Actuator is responsible for performing machine reconciliation
 type IBMCloudClient struct {
-	params ibmcloud.ActuatorParams
-	client client.Client
+	params     ibmcloud.ActuatorParams
+	coreClient corev1.CoreV1Interface
```

Comment:
Any reason to have a new client `coreClient` here?

Reply:
it's used to list nodes, because I have to list all nodes in order to get the node with same name to the machine and return its info such as UID, any other way you know that we can use existing client to list nodes?

Reply:
I think the client is generally for all resources. We may reuse it to list nodes.

Reply:
ok, I am not aware that we can use this client to list nodes, let me try this